### ISSUE TITLE
User Directory Selection Problem

### ISSUE THREAD

**Issue by rachael7** [DA-Osborne/Spool-Maker#2]
On Windows, there's a minor bug in CuraMaterial.py in the section that gets the user directory.  I was missing some materials in the dropdown, and they wouldn't appear when I rescanned the materials, so I did some digging. From a command prompt, I ran CuraMaterial.py and discovered that it was looking at user directory for 4.9, even though I have 4.12 on the machine and it correctly found that for the program directory. I did a bit of diagnostics and it turns out that the sort on the user directory names is alphabetical, since it is working on a string, and that makes the last item in the user directory array not necessarily the most recent, in the case that the user opted to keep their data when uninstalling previous versions.  In my case, I got a sorted user directory list that looked like this:

```
4.1
4.11
4.12
4.4
4.5
4.8
4.9
```

Seeing that, the fix was pretty straightforward, or at least it would have been if I was a Python programmer.  Anyway, all it needs is for this line:

`    CURA_CONFIGS.sort()`

to be replaced with:

`    CURA_CONFIGS.sort(key=lambda x:int(x.replace(".","")))`

And that will get it working all peachy.  I'm using the uncompiled version to work with it now, since that one will call my modified CuraMaterial.py, but I'd love if you could roll in the change and compile a new executable file. I made a fork with the change and put in a pull request, but I've never done that before, so apologies if I didn't do it right. Thanks!

**Comment by rachael7** [comment]
Upon further study, I realize that my fix will break at version 5.0. The proposed fix is perhaps heading the right direction, but it is insufficient as written.

**Comment by rachael7** [comment]
Okay, I think I have an answer that will prove universal now.  The results are right if the sort is done like this:

`CURA_CONFIGS.sort(key=lambda x:int(x.replace(".","" if len(x)>3 else "0")))`

Our list of possible directories includes entries like 4.9 and 4.11, and all versions seem to hold that format. So what I did was, if the directory was 4 characters long (>3), I replaced the decimal point with nothing, leaving 411 instead of 4.11. If the length is 3 or less, I replaced the decimal point with a zero, so 4.9 becomes 409. That string can then be converted to an integer, and as long as Ultimaker doesn't change their directory labeling schema, will correctly sort the list and allow the most recent directory to be used. I tested by making various fictitious version directories within the extant schema, and it correctly chose the highest number version in each case, so I'm pretty confident in the solution.
